src/fakeauth.js

```diff
@@ -0,0 +1,54 @@
+var debug = require('debug')('taskcluster-lib-testing:FakeAuth');
+var nock = require('nock');
+var hawk = require('hawk');
+
+/**
+ * Intercept requests to the auth service's authenticateHawk method and
+ * return a response based on clients, instead.  This is useful when testing
+ * other API services.  Note that accessTokens are not checked -- the fake
+ * simply controls access based on clientId or the scopes in a temporary
+ * credential or supplied with authorizedScopes.
+ *
+ * Clients is on the form
+ * ```js
+ * {
+ *  "clientId1": ["scope1", "scope2"],
+ *  "clientId2": ["scope1", "scope3"],
+ * }
+ *
+ * Call `stop` in your test's `after` method to stop the HTTP interceptor.
+ */
+module.exports.start = function(clients) {
```

Comment:
> Given the overly-aggressive stop we're using right now, I don't want to do something that would pollute other test cases.

Good point, I think if start is called within the suite and passes mocha, then it should work... But fair enough, it's just a neat trick... I agree it might be better to explicit...